### DISCUSSION TITLE
How are you guys using the IPN?

### DISCUSSION THREAD

**Discussion by lookme2** [inventree/InvenTree#2628] [Q&A] [answered]
How do you guys use the IPN?  I was thinking the InvenTree would auto assign this for you?

**Comment by SchrodingersGat** [comment, marked as answer]
How the IPN is used is up to the user - for example many companies have a system for generating "internal part numbers".

Note that the IPN is separate the "unique ID" for the each part, which is internally generated and managed by InvenTree.

**Reply by lookme2** [reply]
So IPN can not be generated via InvenTree?  Could that be a FR?

**Reply by SchrodingersGat** [reply]
What would you be expecting an auto generated ipn to look like?

**Reply by lookme2** [reply]
That I don't know.  It might be nice to be able to give it a prefix and then it act like the PO field may be?

**Reply by eeintech** [reply]
This is a request that has shown up before, automatically generating IPN difficulty lies in implementing a flexible part number builder tool.
Some companies (like ours) use the categories as codes inside the IPN, while others might want to code something else...
The most basic interface could offer user a predefined/fixed prefix and postfix, wrapping the unique ID of the part. In the most basic form the IPN could be equal to the unique ID, maybe with some padding to keep the length the same across IPNs.
If you think this has any value to you, you may open a FR.
One other way to do it is to use the API/Python library to automatically update IPN based on your own scheme.

**Reply by lookme2** [reply]
@eeintech good point.

**Reply by matmair** [reply]
IPNs have a huge meaning for a lot of departments in bigger enterprises. They are often used for routing orders in warehouses, ordering-flows e.g. . So it would be quite hard to find a system that is applicable to both our hobby users and the enterprises. This is a good candidate for a plugin or API script. I would be very interested if you come up with something cool @lookme2 - it is a fascinating problem. Feel free to post in the discussions if you find a solution!
